src/manager.cpp

```diff
@@ -271,17 +271,54 @@ int Manager::updateKeyword(const types::Path i_vpdPath,
         return l_bytesUpdatedOnHardware;
     }
 
-    // Update keyword's value on inventory object path if present
-    if (!l_inventoryObjPath.empty())
+    try
     {
-        int l_bytesUpdatedOnDbus = updateKeywordOnDbus(
-            l_inventoryObjPath, l_fruPath, i_paramsToWriteData);
-
-        if (l_bytesUpdatedOnDbus == -1)
+        // If inventory D-bus object path is present, perform update
+        if (!l_inventoryObjPath.empty())
         {
-            return l_bytesUpdatedOnDbus;
+            types::Record l_recordName;
+            types::DbusInterface l_interfaceName;
+            types::DbusPropertyName l_propertyName;
+            types::DbusVariantType l_keywordValue;
+
+            if (const types::IpzData* l_ipzData =
+                    std::get_if<types::IpzData>(&i_paramsToWriteData))
+            {
+                l_recordName = std::get<0>(*l_ipzData);
+                l_interfaceName = constants::ipzVpdInf + l_recordName;
+                l_propertyName = std::get<1>(*l_ipzData);
+
+                // Read the keyword's value from hardware to write the same on
+                // D-bus.
+                l_keywordValue =
+                    readKeyword(l_fruPath, types::ReadVpdParams(std::make_tuple(
+                                               l_recordName, l_propertyName)));
+            }
+            else
+            {
+                throw std::runtime_error(
+                    "Input parameter type provided isn't compatible to perform update.");
+            }
+
+            // Create D-bus object map
+            types::ObjectMap l_dbusObjMap = {std::make_pair(
+                l_inventoryObjPath,
+                types::InterfaceMap{std::make_pair(
+                    l_interfaceName, types::PropertyMap{std::make_pair(
+                                         l_propertyName, l_keywordValue)})})};
+
+            // Call PIM's Notify method to perform update
+            if (!dbusUtility::callPIM(std::move(l_dbusObjMap)))
+            {
+                throw std::runtime_error("Call to PIM's Notify method failed.");
+            }
         }
     }
+    catch (const std::exception& l_exception)
+    {
+        // TODO: Log PEL
```

Comment:
As there are some string being thrown as explanation about the failure. Should that be logged here along with PEL or just PEL will be ok? (I would like other to weigh in here).
If there is no logging then we don't see anything on journal.

Rest I am ok with the commit.

Reply:
Will do it while creating a PEL.